plugins/woocommerce-blocks/assets/js/blocks/product-template/utils.tsx

```diff
@@ -0,0 +1,158 @@
+/**
+ * External dependencies
+ */
+import { resolveSelect, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { useState, useEffect } from '@wordpress/element';
+// import { store as editorStore } from '@wordpress/editor';
+import { isNumber } from '@woocommerce/types';
+
+type LocationType = 'product' | 'archive' | 'cart' | 'order' | 'generic';
+
+const parseResponse = ( resp?: Record< 'id', number >[] ): number | null =>
+	resp && resp.length && resp[ 0 ]?.id ? resp[ 0 ].id : null;
+const createGetEntitySlug =
```

Comment:
Perhaps we could create a more strict mapper here? We can utilize a global whitelist for template slugs as well in this.

Reply:
It's just an util which seems generic, unless I don't understand your suggestion. Do you mean some validation? Obviously, it's subject to refactoring though!

Also, could you point me to `global whitelist for template slugs`, I'm not sure what you mean by that.

Reply:
> Do you mean some validation? Obviously, it's subject to refactoring though!

Yes. Feel free to skip this comment. I was thinking about adding this block to single-post templates and handling custom-selected context through inspector controls, but we'll handle them later.